### PR TITLE
Add chessx and stockfish (chess engine)

### PR DESCRIPTION
**Pull request by luispedro**
These are utilities to analyse chess games (a viewer, chessx; and a chess engine, stockfish).

The chess engine is actually optional for chessx usage, but I it's a very useful add-on and the whole stockfish package takes on 344K on disk, so I chose to just list it as a `propagatedBuildInput` so users get it automatically.

### CODE REVIEW

**Comment by copumpkin** [comment]
Not sure `propagatedBuildInputs` is the right place to put what you're effectively making into a dependency. How does `chessx` detect the engine at runtime? Is it through the `PATH` or something else? I'd recommend locking that down a bit, if so.

File: pkgs/games/chessx/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, pkgconfig, pkgs, qtbase, qtsvg, qttools, qtmultimedia, fetchurl }:
+pkgs.stdenv.mkDerivation rec {
+  name = "chessx";
```
Comment (the-kenny) [inline pkgs/games/chessx/default.nix]: Please use `<name>-<version>`.

File: pkgs/games/chessx/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, pkgconfig, pkgs, qtbase, qtsvg, qttools, qtmultimedia, fetchurl }:
```
Comment (the-kenny) [inline pkgs/games/chessx/default.nix]: any reason you need `pkgs` here?

File: pkgs/games/chessx/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+   stdenv
+   pkgconfig
+
+   qtbase
+   qtsvg
+   qttools
+   qtmultimedia
+
+
+   pkgs.zlib
```
Comment (peti) [inline pkgs/games/chessx/default.nix]: Please take `zlib` as an argument directly and get rid of the `pkgs` argument.

Also, why all this whitespace in the list?

File: pkgs/games/stockfish/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    url = https://stockfish.s3.amazonaws.com/stockfish-6-src.zip;
+    sha256 = "a69a371d3f84338cefde4575669bd930d186b046a10fa5ab0f8d1aed6cb204c3";
+  };
+  buildPhase = ''
+  cd src
+  make build ARCH=x86-64
+  '';
+  buildInputs = [
+        stdenv
+        pkgs.unzip
```
Comment (peti) [inline pkgs/games/stockfish/default.nix]: Please take zlib as an argument directly and get rid of the pkgs argument.

**Comment by luispedro** [comment]
@copumpkin  I realised that chessx was expecting stockfish because (a long time ago), I configured it to use it as the engine, but it does normally require the user to click on the UI to set it up the first time. Thus,  I have removed mentions of it from the package.

**Comment by the-kenny** [comment]
@luispedro If chessx really benefits from stockfish, we should somehow make it discover it automatically (maybe via a small patch to the source if it doesn't support an environment variable).

File: pkgs/games/chessx/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  '';
+  buildInputs = [
+   stdenv
+   pkgconfig
+
+   qtbase
+   qtsvg
+   qttools
+   qtmultimedia
+
```
Comment (peti) [inline pkgs/games/chessx/default.nix]: Why does this list contain there several empty lines?

**Comment by peti** [comment]
IMHO, the discoverability of `stockfish` can be improved in a separate PR. If the outstanding in-line comments are addressed, then this patch is good to be merged. I would very much like to see `meta` attributes for `homepage`, `license`, and `description` added to both expressions, though.

**Comment by luispedro** [comment]
@peti  Indeed this may even be better handled upstream. I am just finishing testing the new version and will push it in a bit. That one should address the outstanding issues.

**Comment by peti** [comment]
Travis reports errors for the patch.

**Comment by luispedro** [comment]
Sorry, fixed now.

**Pull request merged**